packages/graphql-mini-transforms/tests/webpack.test.ts

```diff
@@ -70,6 +70,14 @@ describe('graphql-mini-transforms/webpack', () => {
     );
   });
 
+  it('has option for custom ID generate function', async () => {
+    const result = await extractDocumentExport(
+      `query Shop { shop { id } }`,
+      createLoaderContext({query: {generateId: () => 'foo'}}),
+    );
+    expect(result).toHaveProperty('id', 'foo');
```

Comment:
It would be nice to have the `generateId` option be a spy so we can check that it was called with what we expect (or do this in a separate function)

Reply:
Done in c8c54ff. It got a bit mucky because `stripIndent` strips trailing whitespace, but the actual value has a trailing newline. Originally just went to concatenate a newline on after `stripIndent`, but ESLint didn't like that (and I didn't much like disabling the warning for that line), so settled on `expect.toStartWith` to skip around the trailing newline issue. Suggestions for better solution welcome.